### ISSUE TITLE
java.io.IOException: Stream closed

### ISSUE THREAD

**Issue by starsyi** [PacificBiosciences/SMRT-Analysis#210]
$ referenceUploader -c -p /data3/xxyi/data/wheat_data/reference/ -n wheat_rep -f wheat_rep.fa 
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
java.io.IOException: Stream closed

Hi,

I am running SMARtanalysis (V2.3) and I getting the following error. Assuming the error has something to do with the data files of this sequence, I tried check the format of sequence,and tried use other sequence of wheat, error still exist, then tried
the analysis with small fraction of this sequence, error disappeared.
Any suggestion on solving this error will be appreciated.

Thanks
starsyi

**Comment by jkaralius** [comment]
Hi @starsyi 

The first three lines are  always there, and are what I see as well.
```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```
The fourth line is one I haven't seen, nor could I reproduce it.
```
java.io.IOException: Stream closed
```

When you say that you "tried the analysis with small fraction of this sequence, error disappeared", does that mean you can use the reference?

Can you run the `referenceUploader` command using the `-verbose` flag? Please post the output.
```
referenceUploader -c -p /data3/xxyi/data/wheat_data/reference/ -n wheat_rep -f wheat_rep.fa -verbose
```

**Comment by starsyi** [comment]
Hi，@jkaralius

thanks for your help.
Yes, when I use a small part of sequence, I can use reference.
Then I run the referenceUploader command using -verbose, the results are as follows:

$ referenceUploader -c -p /data3/xxyi/data/wheat_data/reference/ -n wheat_rep -f wheat_rep.fa -verbose
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader main
INFO: 
********** ReferenceUploader
********** Version: 2.3.0
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader run
INFO: The cmdline is: -c -p /data3/xxyi/data/wheat_data/reference/ -n wheat_rep -f wheat_rep.fa -verbose 
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader run
INFO: Starting Create
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.common.sys.linux.Df exec
FINE: Filesystem         1024-blocks      Used Available Capacity Mounted on
192.168.201.220:/nfs/HZNY1-data3 10759194624 6700243968 3512414208      66% /data3

SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=0% ReferenceId=wheat_rep
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:15:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=4% ReferenceId=wheat_rep
Jun 13, 2015 10:15:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:15:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:16:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=9% ReferenceId=wheat_rep
Jun 13, 2015 10:16:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:16:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:17:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=13% ReferenceId=wheat_rep
Jun 13, 2015 10:17:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:17:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:18:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=18% ReferenceId=wheat_rep
Jun 13, 2015 10:18:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:18:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:19:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=22% ReferenceId=wheat_rep
Jun 13, 2015 10:19:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:19:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:20:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=27% ReferenceId=wheat_rep
Jun 13, 2015 10:20:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:20:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:21:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=31% ReferenceId=wheat_rep
Jun 13, 2015 10:21:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:21:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:22:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=36% ReferenceId=wheat_rep
Jun 13, 2015 10:22:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:22:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:23:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=40% ReferenceId=wheat_rep
Jun 13, 2015 10:23:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:23:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:24:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=45% ReferenceId=wheat_rep
Jun 13, 2015 10:24:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:24:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:25:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=49% ReferenceId=wheat_rep
Jun 13, 2015 10:25:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:25:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:26:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=54% ReferenceId=wheat_rep
Jun 13, 2015 10:26:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:26:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:26:46 AM com.pacbio.secondary.common.sys.linux.Df exec
FINE: Filesystem         1024-blocks      Used Available Capacity Mounted on
192.168.201.220:/nfs/HZNY1-data3 10759194624 6702907392 3509751808      66% /data3

Jun 13, 2015 10:26:46 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader main
SEVERE: Failure to complete cmd.
java.lang.RuntimeException: java.io.IOException: Stream closed
	at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.update(ContigIndexWriter.java:61)
	at java.util.Observable.notifyObservers(Observable.java:159)
	at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.writeByte(FastaIndexWriter.java:274)
	at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.writeBytes(FastaIndexWriter.java:255)
	at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.writeIndexLine(FastaIndexWriter.java:223)
	at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.callWriteLastIndexLineOfContig(FastaIndexWriter.java:303)
	at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.finish(FastaIndexWriter.java:287)
	at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.update(FastaIndexWriter.java:131)
	at java.util.Observable.notifyObservers(Observable.java:159)
	at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.writeByte(MultiSeqFastaWriter.java:212)
	at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.writeByte(MultiSeqFastaWriter.java:200)
	at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.handleSeqHeaderByte(MultiSeqFastaWriter.java:182)
	at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.handleByte(MultiSeqFastaWriter.java:114)
	at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.update(MultiSeqFastaWriter.java:88)
	at java.util.Observable.notifyObservers(Observable.java:159)
	at com.pacbio.secondary.analysis.referenceUploader.io.MultiFileByteBroadcaster.stream(MultiFileByteBroadcaster.java:120)
	at com.pacbio.secondary.analysis.referenceUploader.io.MultiFileByteBroadcaster.start(MultiFileByteBroadcaster.java:53)
	at com.pacbio.secondary.analysis.referenceUploader.CmdCreate.run(CmdCreate.java:101)
	at com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader.run(ReferenceUploader.java:89)
	at com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader.main(ReferenceUploader.java:62)
Caused by: java.io.IOException: Stream closed
	at java.io.BufferedWriter.ensureOpen(BufferedWriter.java:116)
	at java.io.BufferedWriter.write(BufferedWriter.java:141)
	at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.writeByte(ContigIndexWriter.java:124)
	at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.writeByte(ContigIndexWriter.java:118)
	at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.handleSeqHeaderByte(ContigIndexWriter.java:96)
	at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.handleByte(ContigIndexWriter.java:68)
	at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.update(ContigIndexWriter.java:56)
	... 19 more

java.io.IOException: Stream closed

**Comment by starsyi** [comment]
Hi，@jkaralius
thanks for your help.
Yes, when I use a small part of sequence, I can use reference.
Then I run the referenceUploader command using -verbose, the results are as follows:
$ referenceUploader -c -p /data3/xxyi/data/wheat_data/reference/ -n wheat_rep -f wheat_rep.fa -verbose
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader main
INFO: 
********** ReferenceUploader
********** Version: 2.3.0
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader run
INFO: The cmdline is: -c -p /data3/xxyi/data/wheat_data/reference/ -n wheat_rep -f wheat_rep.fa -verbose 
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader run
INFO: Starting Create
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.common.sys.linux.Df exec
FINE: Filesystem         1024-blocks      Used Available Capacity Mounted on
192.168.201.220:/nfs/HZNY1-data3 10759194624 6700243968 3512414208      66% /data3
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=0% ReferenceId=wheat_rep
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:14:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:15:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=4% ReferenceId=wheat_rep
Jun 13, 2015 10:15:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:15:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:16:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=9% ReferenceId=wheat_rep
Jun 13, 2015 10:16:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:16:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:17:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=13% ReferenceId=wheat_rep
Jun 13, 2015 10:17:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:17:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:18:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=18% ReferenceId=wheat_rep
Jun 13, 2015 10:18:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:18:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:19:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=22% ReferenceId=wheat_rep
Jun 13, 2015 10:19:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:19:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:20:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=27% ReferenceId=wheat_rep
Jun 13, 2015 10:20:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:20:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:21:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=31% ReferenceId=wheat_rep
Jun 13, 2015 10:21:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:21:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:22:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=36% ReferenceId=wheat_rep
Jun 13, 2015 10:22:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:22:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:23:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=40% ReferenceId=wheat_rep
Jun 13, 2015 10:23:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:23:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:24:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=45% ReferenceId=wheat_rep
Jun 13, 2015 10:24:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:24:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:25:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=49% ReferenceId=wheat_rep
Jun 13, 2015 10:25:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:25:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:26:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
INFO: Stage=FastaCreation Progress=54% ReferenceId=wheat_rep
Jun 13, 2015 10:26:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.HttpCallback instance
FINE: No callback uri specified, returning null
Jun 13, 2015 10:26:18 AM com.pacbio.secondary.analysis.referenceUploader.monitor.ProgressReportJob execute
FINE: Callback uri was not supplied. No progress will be report to webservices.
Jun 13, 2015 10:26:46 AM com.pacbio.secondary.common.sys.linux.Df exec
FINE: Filesystem         1024-blocks      Used Available Capacity Mounted on
192.168.201.220:/nfs/HZNY1-data3 10759194624 6702907392 3509751808      66% /data3
Jun 13, 2015 10:26:46 AM com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader main
SEVERE: Failure to complete cmd.
java.lang.RuntimeException: java.io.IOException: Stream closed
    at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.update(ContigIndexWriter.java:61)
    at java.util.Observable.notifyObservers(Observable.java:159)
    at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.writeByte(FastaIndexWriter.java:274)
    at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.writeBytes(FastaIndexWriter.java:255)
    at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.writeIndexLine(FastaIndexWriter.java:223)
    at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.callWriteLastIndexLineOfContig(FastaIndexWriter.java:303)
    at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.finish(FastaIndexWriter.java:287)
    at com.pacbio.secondary.analysis.referenceUploader.io.FastaIndexWriter.update(FastaIndexWriter.java:131)
    at java.util.Observable.notifyObservers(Observable.java:159)
    at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.writeByte(MultiSeqFastaWriter.java:212)
    at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.writeByte(MultiSeqFastaWriter.java:200)
    at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.handleSeqHeaderByte(MultiSeqFastaWriter.java:182)
    at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.handleByte(MultiSeqFastaWriter.java:114)
    at com.pacbio.secondary.analysis.referenceUploader.io.MultiSeqFastaWriter.update(MultiSeqFastaWriter.java:88)
    at java.util.Observable.notifyObservers(Observable.java:159)
    at com.pacbio.secondary.analysis.referenceUploader.io.MultiFileByteBroadcaster.stream(MultiFileByteBroadcaster.java:120)
    at com.pacbio.secondary.analysis.referenceUploader.io.MultiFileByteBroadcaster.start(MultiFileByteBroadcaster.java:53)
    at com.pacbio.secondary.analysis.referenceUploader.CmdCreate.run(CmdCreate.java:101)
    at com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader.run(ReferenceUploader.java:89)
    at com.pacbio.secondary.analysis.referenceUploader.ReferenceUploader.main(ReferenceUploader.java:62)
Caused by: java.io.IOException: Stream closed
    at java.io.BufferedWriter.ensureOpen(BufferedWriter.java:116)
    at java.io.BufferedWriter.write(BufferedWriter.java:141)
    at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.writeByte(ContigIndexWriter.java:124)
    at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.writeByte(ContigIndexWriter.java:118)
    at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.handleSeqHeaderByte(ContigIndexWriter.java:96)
    at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.handleByte(ContigIndexWriter.java:68)
    at com.pacbio.secondary.analysis.referenceUploader.io.ContigIndexWriter.update(ContigIndexWriter.java:56)
    ... 19 more
java.io.IOException: Stream closed



yxx285223595@163.com
 
From: Joey Karalius
Date: 2015-06-13 02:28
To: PacificBiosciences/SMRT-Analysis
CC: starsyi
Subject: Re: [SMRT-Analysis] java.io.IOException: Stream closed (#210)
Hi @starsyi 
The first three lines are always there, and are what I see as well.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

The fourth line is one I haven't seen, nor could I reproduce it.
java.io.IOException: Stream closed

When you say that you "tried the analysis with small fraction of this sequence, error disappeared", does that mean you can use the reference?
Can you run the referenceUploader command using the -verbose flag? Please post the output.
referenceUploader -c -p /data3/xxyi/data/wheat_data/reference/ -n wheat_rep -f wheat_rep.fa -verbose
